miden/tests/integration/stdlib/collections/smt.rs

```diff
@@ -0,0 +1,191 @@
+            depth,
+            index,
+        ];
+        let advice_stack = [];
+        let advice_map = [(key.to_owned().into_bytes(), value.to_vec())];
+        build_test!(source, &initial_stack, &advice_stack, store.clone(), advice_map)
+            .expect_stack(&expected_output);
+
+        let source = r#"
+            use.std::collections::smt
+
+            begin
+                exec.smt::get
+            end
+        "#;
+        let initial_stack = [
+            root[0].as_int(),
+            root[1].as_int(),
+            root[2].as_int(),
+            root[3].as_int(),
+            key[0].as_int(),
+            key[1].as_int(),
+            key[2].as_int(),
+            key[3].as_int(),
+        ];
+        let expected_output = [
+            depth,
+            index,
+            value[3].as_int(),
+            value[2].as_int(),
+            value[1].as_int(),
+            value[0].as_int(),
+            key[3].as_int(),
+            key[2].as_int(),
+            key[1].as_int(),
+            key[0].as_int(),
+            root[3].as_int(),
+            root[2].as_int(),
+            root[1].as_int(),
+            root[0].as_int(),
+        ];
+        let advice_stack = [];
+        let advice_map = [(key.to_owned().into_bytes(), value.to_vec())];
+        build_test!(source, &initial_stack, &advice_stack, store, advice_map)
+            .expect_stack(&expected_output);
+    }
+
+    // insert the leaf on the first tier
+    let depth = 16;
+    let index = key[3].as_int() >> 48;
+    let mut store = MerkleStore::new();
+    let node = Rpo256::merge_in_domain(&[key.into(), value.into()], Felt::new(depth)).into();
+    let root = store.add_sparse_merkle_tree(depth as u8, [(index, node)]).unwrap();
+    assert_case(depth, index, key, value, node, root, store);
+
+    // insert the leaf on the second tier
+    let depth = 32;
+    let index = key[3].as_int() >> 32;
+    let mut store = MerkleStore::new();
+    let node = Rpo256::merge_in_domain(&[key.into(), value.into()], Felt::new(depth)).into();
+    let root = store.add_sparse_merkle_tree(depth as u8, [(index, node)]).unwrap();
+    assert_case(depth, index, key, value, node, root, store);
+
+    // insert the leaf on the third tier
+    let depth = 48;
+    let index = key[3].as_int() >> 16;
+    let mut store = MerkleStore::new();
+    let node = Rpo256::merge_in_domain(&[key.into(), value.into()], Felt::new(depth)).into();
+    let root = store.add_sparse_merkle_tree(depth as u8, [(index, node)]).unwrap();
+    assert_case(depth, index, key, value, node, root, store);
+
+    // insert the leaf on the last tier
+    let depth = 64;
+    let index = key[3].as_int();
+    let mut store = MerkleStore::new();
+    let node = Rpo256::merge_in_domain(&[key.into(), value.into()], Felt::new(depth)).into();
+    let root = store.add_sparse_merkle_tree(depth as u8, [(index, node)]).unwrap();
+    assert_case(depth, index, key, value, node, root, store);
```

Comment:
I don't think we need to test depth 64 yet (since this will not be correct in the future anyway) - but we should test retrieval of empty nodes. Specifically, for each test case above (depth 16, 32, 48) we should also try get a node at a key which was not inserted to make sure it works and we get `[ZERO; 4]` returned as the value.

Reply:
Depth `64` is, for now, treated as a normal leaf. We don't really need to remove these tests, as they are important to get some edge cases that could happen now (bit overflow, tree limit, etc)

Reply:
I guess these tests don't hurt - but in my mind, the implementation is incomplete and I wouldn't worry about what happens at depth below 48 until we finish it. So, I probably wouldn't have put them in, but if you'd like to keep them, that's fine too.

Reply:
`64` contains a lot of edge cases, besides the ordered lists. Bitshift, for example, will overflow if we attempt to do a naive `v << depth`. I caught many bugs with such tests; I think it's better to keep them for now, even tho the implementation might change when we introduce ordered lists.